tests/test_core.py

```diff
@@ -183,9 +151,19 @@ def test_initialize_raises_error_on_missing_base_url(monkeypatch):
         prefect_saturn.PrefectCloudIntegration(prefect_cloud_project_name=TEST_PREFECT_PROJECT_NAME)
 
 
-@responses.activate
+def test_initialize_raises_error_on_accessing_properties_if_not_registered():
+    integration = prefect_saturn.PrefectCloudIntegration(
+        prefect_cloud_project_name=TEST_PREFECT_PROJECT_NAME
+    )
+    with raises(RuntimeError, match=prefect_saturn.Errors.NOT_REGISTERED):
+        integration.flow_id
+    with raises(RuntimeError, match=prefect_saturn.Errors.NOT_REGISTERED):
+        integration.flow_version_id
+    with raises(RuntimeError, match=prefect_saturn.Errors.NOT_REGISTERED):
+        integration.image
```

Comment:
Reading over this twice it makes me wonder if we should provide a `integration.is_registered()` or something? But I'm not sure if that would be helpful, or much different from checking if `integration.flow_id` is ! None.

Reply:
I kind of like the way we have it where you have three properties that are only populated by registration (`flow_id`, `flow_version_id`, `image`), and then any code that accesses those will hit this error if you're not registered.

If we had a `.is_registered()` that checked one or all three of these properties, that would be *more* code, in a way.